File: js/src/Ice/UnknownSlicedObject.js
// **********************************************************************
//
// Copyright (c) 2003-2017 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

var Ice = require("../Ice/ModuleRegistry").Ice;
Ice.__M.require(module, ["../Ice/Object", "../Ice/Class"]);
    
var SliceInfo = function()
{
    /**
    * The Slice type ID for this slice.
    **/
    this.typeId = "";

    /**
    * The Slice compact type ID for this slice.
    **/
    this.compactId = -1;

    /**
    * The encoded bytes for this slice, including the leading size integer.
    **/
    this.bytes = [];

    /**
    * The Ice objects referenced by this slice.
    **/
    this.objects = [];

    /**
    * Whether or not the slice contains optional members.
    **/
    this.hasOptionalMembers = false;

    /**
    * Whether or not this is the last slice.
    **/
    this.isLastSlice = false;
};
Ice.SliceInfo = SliceInfo;

var SlicedData = function(slices)
{
    this.slices = slices;
};
Ice.SlicedData = SlicedData;

var UnknownSlicedObject = Ice.Class(Ice.Object,
    {
        __init__: function(unknownTypeId)
        {
            this._unknownTypeId = unknownTypeId;
        },
        getUnknownTypeId: function()
        {
            return this._unknownTypeId;
        },
        __write: function(os)
        {
            os.startWriteObject(this._slicedData);
            os.endWriteObject();
        },
        __read: function(is)
        {
            is.startReadObject();
            this._slicedData = is.endReadObject(true);
        }
    });
Ice.UnknownSlicedObject = UnknownSlicedObject;

module.exports.Ice = Ice;

File: csharp/src/Ice/UnknownSlicedObject.cs
// **********************************************************************
//
// Copyright (c) 2003-2017 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

namespace Ice
{
    /// <summary>
    /// Unknown sliced object holds an instance of unknown type.
    /// </summary>
    public sealed class UnknownSlicedObject : ObjectImpl
    {
        /// <summary>
        /// Instantiates the class for an Ice object having the given Slice type.
        /// </summary>
        /// <param name="unknownTypeId">The Slice type ID of the unknown object.</param>
        public UnknownSlicedObject(string unknownTypeId)
        {
            _unknownTypeId = unknownTypeId;
        }

        /// <summary>
        /// Determine the Slice type ID associated with this object.
        /// </summary>
        /// <returns>The type ID.</returns>
        public string getUnknownTypeId()
        {
            return _unknownTypeId;
        }

        public override void write__(IceInternal.BasicStream os__)
        {
            os__.startWriteObject(_slicedData);
            os__.endWriteObject();
        }

        public override void read__(IceInternal.BasicStream is__)
        {
            is__.startReadObject();
            _slicedData = is__.endReadObject(true);
        }

        private string _unknownTypeId;
        private SlicedData _slicedData;
    }
}
